src/main/java/org/folio/clients/InventoryClient.java

```diff
@@ -0,0 +1,62 @@
+package org.folio.clients;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.folio.util.OkapiConnectionParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+@Component
+public class InventoryClient {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private static final String GET_INSTANCES_URL = "%s/instance-storage/instances?query=(%s)";
+  private static final String QUERY_PATTERN_INVENTORY = "id==%s";
+  private static final String LIMIT_PATTERN = "&limit=";
+
+  private static final int SETTING_LIMIT = 100;
+  private static final String NATURE_OF_CONTENT_TERMS_URL = "%s/nature-of-content-terms?limit=" + SETTING_LIMIT;
+
+  public Optional<JsonObject> getInstancesByIds(List<String> ids, OkapiConnectionParams params, int partitionSize) {
+    return ClientUtil.getByIds(ids, params, GET_INSTANCES_URL + LIMIT_PATTERN + partitionSize, QUERY_PATTERN_INVENTORY);
+  }
+
+  public List<JsonObject> getNatureOfContentTerms(OkapiConnectionParams params) {
+    return getSettingsByUrl(NATURE_OF_CONTENT_TERMS_URL, params, "natureOfContentTerms");
```

Comment:
I think "natureOfContentTerms" could be static final variable, but it is not a problem if you will leave it as it is now